### ISSUE TITLE
Why is Python awesome?

### ISSUE THREAD

**Issue by tommeagher** [ireapps/pycar#28]
Again, this is feedback [from students last year](https://github.com/ireapps/pycar/issues/17#issuecomment-77927152). We do go into this in the introduction, but are there other examples we should use or points we should be sure to hit later in the day?

**Comment by richardsalex** [comment]
This isn't a specific example, but I've asked a couple of classes to think about a really repetitive part of their reporting workday or week. The things that end up eating up hours of time when you add it all up, like collecting information from a spreadsheet that gets updated regularly or a map they're trying to keep current online but they have to periodically punch in values by hand — stuff like that. That's my reason for Python being awesome: I can dump repetitive, mundane on the computer to have more time to work on things I wouldn't otherwise have time to do or get stories I wouldn't otherwise be able to get. My first Python script was this hacky thing that periodically scraped a county recorder's website for distressed properties when there was no other way to get them. The county's position was that by making the records available online, they fulfilled their obligations under open records law completely.

**Comment by zufanka** [comment]
Good point! The epiphany of  "I'm actually doing something repetitive that
can be delegated to a computer" is still something many don't realize.

Once I was working on a quite big investigative project with 4 other
(completely nontechnical) people. At one meeting, I could not attend, they
have decided to create a spreadsheet of all side jobs of Dutch professors.
There are 12 universities and each has approximately 500 professors (you do
the math). They have then divided the universities between us and send them
around.
A week has passed and a day before the next meeting I have panicked, as I
still didn't get the uni assigned to me done. So in a couple of hours wrote
a simple scraper/ parser that could easily fill the spreadsheet with about
2000 rows (one job per row, as it was flat data and each professor could
have multiple side jobs). The next day I have arrived at the newsroom and
told my colleagues I have finished the job. They were looking at me
weirdly. "Wait, what? How?". I was confused. I took a peak into their
spreadsheets: none of them was further than 50 rows. Then I realized (in
horror mixed with amusement) that they were clicking through the websites
and copy-pasting everything. There goes your precious investigative
reporter time.
To be fair: In the end they did quite some hours of data cleaning, but that
time was still only a fraction of what it would cost to do everything
manually.


On Fri, Jan 29, 2016 at 8:06 PM, Alex Richards <notifications@github.com>
wrote:

> This isn't a specific example, but I've asked a couple of classes to think
> about a really repetitive part of their reporting workday or week. The
> things that end up eating up hours of time when you add it all up, like
> collecting information from a spreadsheet that gets updated regularly or a
> map they're trying to keep current online but they have to periodically
> punch in values by hand — stuff like that. That's my reason for Python
> being awesome: I can dump repetitive, mundane on the computer to have more
> time to work on things I wouldn't otherwise have time to do or get stories
> I wouldn't otherwise be able to get. My first Python script was this hacky
> thing that periodically scraped a county recorder's website for distressed
> properties when there was no other way to get them. The county's position
> was that by making the records available online, they fulfilled their
> obligations under open records law completely.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ireapps/pycar/issues/28#issuecomment-176915592>.
>



-- 

Adriana Homolova

data journalist

+31-652245912 | 0F097E0F | @naberacka
